src/api/grainConfig.js

```diff
@@ -39,6 +49,17 @@ export function toDistributionPolicy(x: GrainConfig): DistributionPolicy {
       policyType: "IMMEDIATE",
     });
   }
+  if (x.recentPerWeek > 0) {
+    const discount = NullUtil.orElse(x.discount, undefined);
+    if (discount === undefined) {
+      throw new Error(`no discount specified for recent policy`);
+    }
```

Comment:
No need for NullUtil.orElse in this case, `if (discount == null)` is fine. Note that `undefined == null` and `null == null` are both true. It's the one acceptable use for the `==` operator.